driver/topology_service.cc

```diff
@@ -186,6 +183,48 @@ std::shared_ptr<CLUSTER_TOPOLOGY_INFO> TOPOLOGY_SERVICE::get_topology(CONNECTION
     return cached_topology;
 }
 
+std::shared_ptr<CLUSTER_TOPOLOGY_INFO> TOPOLOGY_SERVICE::get_filtered_topology(std::shared_ptr<CLUSTER_TOPOLOGY_INFO> topology) {
+    if (!this->allowed_and_blocked_hosts) {
+        return topology;
+    }
+
+    std::set<std::string> allowed_list = this->allowed_and_blocked_hosts->get_allowed_host_ids();
+    std::set<std::string> blocked_list = this->allowed_and_blocked_hosts->get_blocked_host_ids();
+
+    const std::shared_ptr<CLUSTER_TOPOLOGY_INFO> filtered_topology = std::make_shared<CLUSTER_TOPOLOGY_INFO>();
+    if (allowed_list.size() > 0) {
+        for (const auto& host : topology->get_instances()) {
+            if (allowed_list.find(host->get_host_id()) != allowed_list.end()) {
+                filtered_topology->add_host(host);
+            }
+        }
+    }
+
+    if (blocked_list.size() > 0) {
+        for (const auto& host : filtered_topology->get_instances()) {
+            // Remove blocked hosts from the filtered_topology.
+            if (blocked_list.find(host->get_host_id()) != blocked_list.end()) {
+                filtered_topology->remove_host(host);
+            }
+        }
+    }
```

Comment:
What's the purpose of this section? Thought the `ALLOWED_AND_BLOCKED_HOSTS` only allowed either one (allow) or the other (blocked) to be set. I don't see this being set explicitly in code being that case but that is how its currently being used and how its being fetched by the AWS API. Going with that, if you have a blocked list, you have no allowed list, so you would never have anything in your `filtered_topology`. Did you mean to get all instances instead of the filtered_topology and filter the block list from there?